packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -775,6 +775,79 @@ export class Stack extends CoreConstruct implements ITaggable {
     }
   }
 
+  /**
+   * Manually create a CloudFormation Export for a Resource attribute
+   *
+   * If you use Resource attributes in another Stack, those CloudFormation
+   * Exports will be created automatically, so usually you won't need to use
+   * this method.
+   *
+   * The only time you need to call this method if you want to *remove* the
+   * relationship between two Stacks. It will temporarily ensure that the CloudFormation
+   * Export still exists while you remove the reference from the consuming stack.
+   * After that, you can remove the resource and the manual export.
+   *
+   * ## Example
+   *
+   * Here is how the process works. Let's say there are two stacks,
+   * `producerStack` and `consumerStack`, and `producerStack` has a bucket
+   * called `bucket`, which is referenced by `consumerStack` (perhaps because
+   * an AWS Lambda Function writes into it, or something like that).
+   *
+   * It is not safe to remove `producerStack.bucket` because as the bucket is being
+   * deleted, `consumerStack` might still be using it.
+   *
+   * Instead, the process takes two deployments:
+   *
+   * ### Deployment 1: break the relationship
+   *
+   * - Make sure `consumerStack` no longer references `bucket.bucketName` (maybe the consumer
+   *   stack now uses its own bucket, or it writes to an AWS DynamoDB table, or maybe you just
+   *   remove the Lambda Function altogether).
+   * - In the `ProducerStack` class, call `this.exportAttribute(this.bucket.bucketName)`. This
+   *   will make sure the CloudFormation Export continues to exist while the relationship
+   *   between the two stacks is being broken.
+   * - Deploy (this will effectively only change the `consumerStack`, but it's safe to deploy both).
+   *
+   * ### Deployment 2: remove the bucket resource
+   *
+   * - You are now free to remove the `bucket` resource from `producerStack`.
+   * - Don't forget to remove the `exportAttribute()` call as well.
+   * - Deploy again (this time only the `producerStack` will be changed -- the bucket will be deleted).
+   */
+  public exportAttribute(exportedAttribute: any) {
```

Comment:
As discussed, I am wondering if we can make this slightly more generic and resolve https://github.com/aws/aws-cdk/issues/2036 along the way:

```ts
const importValue = stack.export(attribute); // auto-name
const importValue = stack.export(1234, { name: 'my-export-name' }); // since the value is not an attribute `name` is required
```